### DISCUSSION TITLE
I get a build error: unknown error, open 'C:\Users\ADMINI~1\AppData\Local\Temp\tmpy9d8kfsz.js'

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#22148] [Q&A]
emcc: 3.1.55

I get an unknown error by executing this emcc command:

`C:\emsdk\upstream\emscripten\\emcc.bat Emscripten/Debug/TEApiWrapper81_for_emscripten.o Emscripten/Debug/Main.o --pre-js pre-js.js -ljpeg.Emscripten -lxml2 -lprjplg.Emscripten -lproj -lexpat -lgdal -lgeos -lgeos_c -l3ds.Emscripten -lTerraExplorerX.Emscripten -lCore.Emscripten -l7zip.Emscripten -lzip.Emscripten -llibzip.Emscripten -lgisplg.Emscripten -lMptDll.Emscripten -lOpenGL.Emscripten -lPlgShell.Emscripten -lSLLogger.Emscripten -lSLLoggerLib.Emscripten -lSLStrings.Emscripten -lTerraCommon.Emscripten -lTerra.Emscripten -lTESWQ.Emscripten -lWFSPlugin.Emscripten -lOGRPlugin.Emscripten -lsl_png.Emscripten -ldraco.Emscripten -lOpenCTM.Emscripten -lwebp.Emscripten -lMPTPlg.Emscripten -lOGRSpatialReference244.Emscripten -s ALLOW_TABLE_GROWTH -s WASM_BIGINT  -s EXPORTED_RUNTIME_METHODS=ccall,cwrap,addFunction,allocateUTF8   -s EXPORTED_FUNCTIONS=_main  -g  -gsplit-dwarf -gdwarf-5 -gpubnames --bind -s USE_PTHREADS -s ALLOW_MEMORY_GROWTH=1    -s ASSERTIONS  -s FETCH   -s TEXTDECODER=0  -lidbfs.js -s PTHREAD_POOL_SIZE=GetThreadPoolSize() -s ERROR_ON_WASM_CHANGES_AFTER_LINK -s REVERSE_DEPS=all -s STACK_SIZE=5mb -s DEFAULT_PTHREAD_STACK_SIZE=5mb -s ALLOW_BLOCKING_ON_MAIN_THREAD=1  -s PTHREAD_POOL_SIZE_STRICT=2  -fwasm-exceptions  -s MAXIMUM_MEMORY=700mb -O0 --preload-file com.skyline.terraexplorer -o C:/Projects/TerraExplorer_v8_0_0_Dev/SLSource/Emscripten/Debug/TEF/TerraExplorerApp.Emscripten.js -LC:/Projects/TerraExplorer_v8_0_0_Dev/SLSource/Emscripten/Debug/ -LC:/Projects/TerraExplorer_v8_0_0_Dev/SLSource/Emscripten/TerraExplorerApp.Emscripten/libs
`

Error:
```
EMSCRIPTENLINK : error : UNKNOWN: unknown error, open 'C:\Users\ADMINI~1\AppData\Local\Temp\tmpy9d8kfsz.js'
emcc : error : 'C:/emsdk/node/18.20.3_64bit/bin/node.exe C:\emsdk\upstream\emscripten\src\compiler.mjs C:\Users\ADMINI~1\AppData\Local\Temp\tmp2hcb8ton.json' failed (returned 1)
```

The path C:\Users\ADMINI~1\AppData\Local\Temp\tmpy9d8kfsz.js is not exist, don't know what is use for. 

Clean & Rebuild doesn't help.
git pull and install from emsdk also doesn't help

**Comment by kripken** [comment]
That's a temp file that emscripten creates as it runs. If there is a problem creating or reading such a file, maybe there is some kind of permissions issue with that temp folder on your machine (if so, I'm not sure how to fix that as I'm not familiar with Windows; on Linux I would try to delete the temp folder).

**Reply by sbc100** [reply]
I'm a little confused by the fact that its `compiler.mjs` that is failing.  As far as I know `compiler.mjs` writes it output the stdout and does not create any temp files at all.

I'm also wondering where the string `EMSCRIPTENLINK` comes from?  That isn't a string we have anywhere in emscripten so I guess its coming from your build system maybe?

As @kripken says it looks like a problem with you files.. are you running your build as the admin user?  

If you can't figure out what is going on perhaps you could run the same command with `EMCC_DEBUG=1` set in the environment that share the output of that command here.

**Comment by aharondavid** [comment]
Yes , you right, it is Permission issue.
I had to insert the Temp folder to the Windows Defender.
